Some D2k clean-ups

reaperrr opened the pull request:
Adresses most remaining issues from  #7274.

reaperrr committed:
Add Mercenaries and Smugglers to D2k

Rename some structures.

reaperrr committed:
Some misc D2k cleanup

C. Yard doesn't need DamagedWithoutFoundation, aircraft aren't selectable so they don't need a voice, ordos tank ROT matches default, RepairBuildings on carryall is redundant.

reaperrr commented:
Hm, yeah. I think Smugglers and Mercs were using Ordos graphics for everything else, right?

Mailaender commented:
I think the roof for the heavy factory is different, but otherwise all graphics are identical. Look at https://d2kplus.com/map-and-mission-editor-by-klofkac/ to be sure.

reaperrr commented:
Updated the merc/smuggler commit.

They now use ordos graphics for everything except their exclusive faction images (starport for smugglers, heavy factory for mercs).

Additionally, removed bogus, potentially crashing `corrino: conyard.corrino` entry (`conyard.corrino` sequence doesn't exist, corrino uses harkonnen gfx anyway).

pchote commented on mods/d2k/rules/structures.yaml:
```diff
@@ -79,7 +80,8 @@ construction_yard:
 			ordos: conyard.ordos
-			corrino: conyard.corrino
+			smuggler: conyard.ordos
+			mercenaries: conyard.ordos
```
You could make these definitions more efficient by making the default `Image: conyard.ordos` (and so on for other buildings), and then you only need to override `harkonnen: conyard.harkonnen` instead of overriding ordos, smugger, and mercinary.

reaperrr replied:
I noticed, but 
a) I'd have to override corrino as well, so we'd save only 1 line per structure and not 2, and 
b) it would increase the diff and changing the default increases the risk of regressions.

I'll do it if you insist, but I felt it wasn't worth the additional work.

pchote replied:
Nope, not insisting.

pchote commented on mods/d2k/rules/world.yaml:
```diff
@@ -35,6 +35,14 @@
+		Selectable: false
+	Faction@Smugglers:
+		Name: Smugglers
+		InternalName: smuggler
```
Internal name `smuggler` (singular) vs `mercinaries` (plural) seems a bit odd.  Is this correct?

MustaphaTR replied:
No, it should be `smugglers`.

reaperrr replied:
In the original game the internal name for smugglers was `smuggler`, so I used that, but I couldn't find out the internal name for the mercs.

I now looked at the d2k exe a moment ago and the internal name for mercs is indeed singular `mercenary` as well.

Does it make a difference to the map importer? If yes, I'd change `mercenaries` to `mercenary`, if no, I'd change `smuggler` to `smugglers`.

pchote replied:
Lets go with `mercenary`.

MustaphaTR replied:
Plural ones sound better but if orginal game used singular ones go with singulars.

reaperrr commented:
Updated, changed internal name from `mercenaries` to `mercenary` to match original.

reaperrr commented:
Updated, moved the weapons.yaml split to its own PR.

pchote commented on mods/d2k/rules/misc.yaml:
```diff
@@ -265,7 +265,7 @@ upgrade.heavy:
 			atreides: heavy.atreides
 			ordos: heavy.ordos
-			corrino: heavy.corrino
+			mercenary: heavy.mercenary
```
This probably wants a `smugger` definition too for consistency.

pchote commented:
:+1: after that small addition.

reaperrr committed:
Fix faction-specific upgraded factory images for mercenaries and smugglers

reaperrr commented:
Updated, fixed upgraded merc/smuggler barracks and light factory images as well.

obrakmann commented:
Looks ok to me :+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/9a139aa86b4087337317354f733d3a5ef20fbcaa)
